wormhole-connect/src/views/v2/Bridge/AmountInput/index.tsx

```diff
@@ -54,13 +54,17 @@ const DebouncedTextField = memo(
         }
 
         setInnerValue(e.target.value);
-        defferedOnChange(e.target.value);
+        deferredOnChange(e.target.value);
       },
       [],
     );
 
+    // Propagate any outside changes to the inner TextField value.
+    // Please note that we need to compare Number values, because '' and 0 are the same in terms of the amount value.
     useEffect(() => {
-      setInnerValue(value);
+      if (Number(innerValue) !== Number(value)) {
```

Comment:
can we please stop using Number for parsing/comparing amounts, everywhere? I just refactored out all the code that did this. it probably doesn't matter here, but I just don't want to set that precedent again. if we have to parse and compare amounts, we should use amount.Amount.

in this case, I don't think we need to do this at all. I think the data flow in this `AmountInput` / `DebouncedTextField` could be simplified. I believe the only debounce we need on input change is on the `dispatch` to Redux, because it's what triggers quote fetching. we shouldn't be debouncing the inner state updates at all. essentially, any keystroke or "Max" button push should update the state immediately and we would only debounce the call to `dispatch`.

Reply:
About the Number() parsing: This is for a unit-less comparison user entered values, not saved or displayed anywhere and won't cause any issues. That being said, I'm not against your recommendation for integrity in the code base 👍 

About the data flow: The amount from redux state has to be propagated to the local state that keeps the input control value at some point. We set the initial value for the state variable; but when user swaps the assets, the amount in redux is cleared and so the state var that keeps the amount input value. That's why we need the side-effects. I'm not a big fan of nested components to handle the debounce, but I can look into that separately to further simplify it.